types/fs-base-ui/html_ui/JS/Avionics.d.ts

```diff
@@ -0,0 +1,198 @@
+
+            allTexts: SVGTextElement[];
+            posX: number;
+            posY: number;
+            spacing: number;
+
+            construct(parent: Node, posX: number, posY: number, width: number, fontFamily: string,
+                      fontSize: number, fontColor: string): void;
+            clear(value: string): void;
+            update(value: number, divider: number, hideIfLower: number): void;
+        }
+
+        class AltitudeScroller extends Scroller {
+            constructor(nbItems: number, spacing: number, increment: number, moduloValue: number, notched: number);
+
+            allTexts: SVGTextElement[];
+            posX: number;
+            posY: number;
+            spacing: number;
+
+            construct(parent: Node, posX: number, posY: number, width: number, fontFamily: string,
+                      fontSize: number, fontColor: string): void;
+            clear(value: string): void;
+            update(value: number, divider: number, hideIfLower: number): void;
+        }
+
+        class SVGArc {
+            static radiansPerPercent: number;
+
+            percent: number;
+            angle: number;
+            gaugeCenterX: number;
+            gaugeCenterY: number;
+            gaugeCloseThreshold: number;
+            computedStartX: number;
+            computedStartY: number;
+            computedEndX: number;
+            computedEndY: number;
+            computedEndAngle: number;
+
+            arc?: HTMLElement;
+            arcRadius?: number;
+            arcSize?: number;
+            arcColor?: string;
+
+            init(arcName: string, radius: number, size: number, color: string): void;
+            setPercent(percent: number): void;
+            translate(x: number, y: number): void;
+            rotate(angle: number): void;
+            get svg(): HTMLElement;
+            get centerX(): number;
+            get centerY(): number;
+            get radius(): number | undefined;
+            get startX(): number;
+            get startY(): number;
+            get endX(): number;
+            get endY(): number;
+            get startAngle(): number;
+            get endAngle(): number;
+            get currentPercent(): number;
+            updateShape(): void;
+            getArcPath(percent: number, radius: number): string;
+        }
+
+        class CurveTimeValuePair {
+            constructor(time: number, value: any);
+
+            time: number;
+            value: any;
+        }
+
+        class CurveTool {
+            static NumberInterpolation(n1: number, n2: number, dt: number): number;
+            static StringColorRGBInterpolation(c1: string, c2: string, dt: number): string;
+        }
+
+        class Curve {
+            add(time: number, value: any): void;
+            evaluate(time: number): number | string;
+            debugLog(): void;
+        }
+
+        class DictionaryItem {
+        }
+
+        class Dictionary {
```

Comment:
The problem with this `Dictionary` type is that it isn't a dictionary. A dictionary in JavaScript would in fact be an object, where the object's properties would be the key and thus have either type `string | number` and the object's value would then be any type. This would be far more efficient. I.e.:
```typescript
let innerObj = {};

function set(key: string | number, value: any) {
    innerObj[key] = value;
}

function get(key: string | number): any {
    return innerObj[key];
}
```

Nothing in the code tells me it is limited to a particular type of key or value so any is best. **I'd definitely not use it in our code.**

Reply:
Yes, definitely. Also, the only addition of `Dictionary` over a proper [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), which could have arbitrary keys and still has constant time access, would be the `changed` property. If `changed` is indeed useful (and therefore this Dictionary class), we could just replace the `items` array with a Map and implement `set`, `get`, `remove` and `exists` in terms of this. But we aren't using `Dictionary` anyway.